### PR TITLE
Added suit-generator payload_extract command

### PR DESCRIPTION
**Pull request by ahasztag**
The payload allows for extracting/removing/replacing integrated payloads in the envelope.

Ref: NCSDK-29708

### CODE REVIEW

File: suit_generator/cmd_payload_extract.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    cmd_payload_extract_arg_parser.add_argument(
+        "--payload-name",
+        required=True,
+        help="Name of the integrated payload to extract."
+    )
+    cmd_payload_extract_arg_parser.add_argument("--output-payload-file", required=False, help="Output payload file path to store the extracted payload."
+                                                + "If not provided, the payload will not be stored to a file.")
+
+    cmd_payload_extract_arg_parser.add_argument(
+        "--payload-replace-name",
```
Comment (tomchy) [inline suit_generator/cmd_payload_extract.py:38]: should not that be `path` not `name`?
I find it difficult to understand this arg description.

**Review by adsz-nordic — APPROVED** [review]

File: suit_generator/cmd_payload_extract.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    :param input_envelope: input envelope file path
+    :param output_envelope: output envelope file path
+    :param payload_name: name of the integrated payload to extract
+    :param output_payload_file: output file path to store the extracted payload
+        None if the payload should not be stored to a file
+    :param payload_replace_name: name of the integrated payload to replace the extracted payload with.
+        None if the payload should be removed from the envelope.
+    """
+    envelope = None
```
Comment (robertstypa) [inline suit_generator/cmd_payload_extract.py:54]: The `envelope` value is immediately assigned in line 56 as `envelope = cbor2.load(fh)`, so the initial assignment is redundant and could be removed

File: suit_generator/cmd_payload_extract.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    with open(input_envelope, "rb") as fh:
+        envelope = cbor2.load(fh)
+    extracted_payload =  envelope.value.pop(payload_name, None)
+
+    if extracted_payload == None:
```
Comment (robertstypa) [inline suit_generator/cmd_payload_extract.py:59]:
```suggestion
    if extracted_payload is None:
```

File: suit_generator/cmd_payload_extract.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        log.log(logging.ERROR, 'Payload "%s" not found in envelope', payload_name)
+
+    if payload_replace_name != None:
```
Comment (robertstypa) [inline suit_generator/cmd_payload_extract.py:62]:
```suggestion
    if payload_replace_name is not None:
```

File: suit_generator/cmd_payload_extract.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        with open(payload_replace_name, "rb") as fh:
+            envelope.value[payload_name] = fh.read()
+
+    with open(output_envelope, "wb") as fh:
+        cbor2.dump(envelope, fh)
+    if output_payload_file != None:
```
Comment (robertstypa) [inline suit_generator/cmd_payload_extract.py:68]:
```suggestion
    if output_payload_file is not None:
```

**Review by robertstypa — CHANGES_REQUESTED** [review]

File: suit_generator/cmd_payload_extract.py (hunk 2 above)
Comment (ahasztag) [reply suit_generator/cmd_payload_extract.py]: Fixed

File: suit_generator/cmd_payload_extract.py (hunk 3 above)
Comment (ahasztag) [reply suit_generator/cmd_payload_extract.py]: Fixed

File: suit_generator/cmd_payload_extract.py (hunk 4 above)
Comment (ahasztag) [reply suit_generator/cmd_payload_extract.py]: Fixed

**Review by robertstypa — APPROVED** [review]

**Review by SylwesterKonczyk — APPROVED** [review]

File: suit_generator/cmd_payload_extract.py (hunk 1 above)
Comment (ahasztag) [reply suit_generator/cmd_payload_extract.py]: You're absolutely right, should be path, changed this

**Pull request merged**